### PR TITLE
App modifications in database (schema and testdata)

### PR DESCRIPTION
**Pull request by ckabelin**
- Schema addition: urls for app- and marketing-endpoint
- Testdata modification according to:
[
  {
    "id": "",
    "title": "BPDM Services",
    "subtitle": "CDQ",
    "description": "Business Partner Services for Fraud & Data Management",
    "price": "free of charge",
    "useCases": [ "BPDM" ],
    "url": "https://apps.cdq.com/signin/catenax",
    "leadPicture": ""
  },
  {
    "id": "ac1cf001-7fbc-1f2f-817f-bce0572c0007",
    "title": "Part Chain",
    "subtitle": "Catena-X",
    "description": "CX App Part Chain Details",
    "price": "free of charge",
    "useCases": [ "Traceability" ],
    "url": "",
    "leadPicture": ""
  },
  {
    "id": "ac1cf001-7fbc-1f2f-817f-bce0573f0009",
    "title": "Dismantler App",
    "subtitle": "SAP",
    "description": "SAP App Dismantler App Details",
    "price": "free of charge",
    "useCases": [ "Circular Economy" ],
    "url": "https://catenax-dt-rec.authentication.eu10.hana.ondemand.com/login",
    "leadPicture": ""
  },
  {
    "id": "ac1cf001-7fbc-1f2f-817f-bce05744000b",
    "title": "CE Marketplace",
    "subtitle": "SAP",
    "description": "SAP App CE Marketplace Details",
    "price": "free of charge",
    "useCases": [ "Circular Economy" ],
    "url": "https://catenax-dt-rec.authentication.eu10.hana.ondemand.com/login",
    "leadPicture": ""
  },
  {
    "id": "ac1cf001-7fbc-1f2f-817f-bce05748000d",
    "title": "Material Traceability",
    "subtitle": "Catena-X",
    "description": "SAP App Material Traceability Details",
    "price": "free of charge",
    "useCases": [ "Traceability" ],
    "url": "",
    "leadPicture": ""
  },
  {
    "id": "ac1cf001-7fbc-1f2f-817f-bce0574c000f",
    "title": "Component Performance",
    "subtitle": "Catena-X",
    "description": "BMW App Component Performance Monitor Details",
    "price": "free of charge",
    "useCases": [ "Quality & Demand Management" ],
    "url": "https://impact.bmw.cloud/workspace/carbon/ri.carbon.main.workspace.7d7f6c71-3981-4b78-b731-1b4c8f243c97/ri.workshop.main.module.c9beba25-4387-40dd-9e59-8c4e3b3df3ff",
    "leadPicture": ""
  },
  {
    "id": "1a4ac107-2b6e-4190-89df-4b8442b12f05",
    "title": "Covanto - AFQM",
    "subtitle": "BOSCH",
    "description": "Automatic monitoring (Early Warning System) of diagnostic data",
    "price": "free of charge",
    "useCases": [ "Quality & Demand Management" ],
    "url": "https://portal-staging.afqm-services.com/[#/auth/signin]",
    "leadPicture": ""
  },
  {
    "id": "1a4ac107-2b6e-4190-89df-4b8442b12f05",
    "title": "Test MultiApp",
    "subtitle": "Catena-X",
    "description": "Works with multiple use cases",
    "price": "free of charge",
    "useCases": [ "Traceability", "Quality & Demand Management" ],
    "url": "",
    "leadPicture": ""
  }
]

### CODE REVIEW

File: postgres/init/22-portal-content.sql [hunk 1]
```diff
@@ -208,7 +225,10 @@ COPY portal.app_descriptions (date_created, date_last_changed, description_long,
...
+2022-03-24 18:01:33.388	2022-03-24 18:01:33.388	Business Partner Services for Fraud & Data Management	Business Partner Services for Fraud & Data Management	f9cad59d-84b3-4880-a550-4072c26a6b93	en
```
Comment (ntruchsess) [inline postgres/init/22-portal-content.sql:229]: That line does not import:
ERROR:  invalid byte sequence for encoding "UTF8": 0x92
CONTEXT:  COPY app_descriptions, line 5
(I guess it's the '&' that needs propper encoding.

**Review by ntruchsess — CHANGES_REQUESTED** [review]
those changes do not import properly, see my comments that refer to the postgres error-messages

File: postgres/init/22-portal-content.sql [hunk 2]
```diff
@@ -95,8 +102,8 @@ ac1cf001-7fbc-1f2f-817f-bce0574c000f	2022-03-24 18:01:33.388	2022-03-24 18:01:33
 COPY portal.use_cases (use_case_id, date_created, date_last_changed, name, shortname) FROM stdin;
 ac1cf001-7fbc-1f2f-817f-bce056f90000	2022-03-24 18:01:33.297	2022-03-24 18:01:33.297	Circular Economy	CE
 ac1cf001-7fbc-1f2f-817f-bce056fa0001	2022-03-24 18:01:33.306	2022-03-24 18:01:33.306	Traceability	Traceability
-ac1cf001-7fbc-1f2f-817f-bce056fa0002	2022-03-24 18:01:33.306	2022-03-24 18:01:33.306	Quality Management	QM
```
[the remark this answers is not in the document]
Comment (ckabelin) [reply postgres/init/22-portal-content.sql:98]: The above testdata mentions a new Use Case "Quality & Demand Management" for the testdata, hence I removed the QM & DM and their associations accordingly from the content. Does the error also occur on a newly instanciated db instance? My testimports on a fresh db went through with the suggested file.

[the remark this answers is not in the document]
Comment (ckabelin) [reply postgres/init/22-portal-content.sql:98]: Has been resolved with the most recent commit.

File: postgres/init/22-portal-content.sql (hunk 1 above)
Comment (ckabelin) [reply postgres/init/22-portal-content.sql:229]: Has been resolved with the most recent commit. Local tests were fine without errors.

**Pull request merged**